Framework/CurveFitting/src/Functions/Bk2BkExpConvPV.cpp

```diff
@@ -181,60 +183,13 @@ double Bk2BkExpConvPV::calOmega(double x, double eta, double N, double alpha, do
   if (eta < 1.0E-8) {
     omega2 = 0.0;
   } else {
-    omega2 = 2 * N * eta / M_PI * (imag(exp(p) * E1(p)) + imag(exp(q) * E1(q)));
+    omega2 = 2 * N * eta / M_PI * (imag(exp(p) * exponentialIntegral(p)) + imag(exp(q) * exponentialIntegral(q)));
```

Comment:
I seem to be getting some unusual results for some typical parameters for ENGIN-X - here I compare what I get on this PR with my dev-branch (31691_Bk2BkConvPV_parameters_for_ENGINX) which includes your implementation (before using `SpecialFunctionSupport`) - albeit with some other changes I have made. Could you test again to make sure E1 is the same for both implementations?
![image](https://user-images.githubusercontent.com/55979119/123250773-5e04c380-d4e2-11eb-9ec9-40f66e9e2456.png)

Reply:
You only see a problem on your branch when `gamma > 0` - so definitely looking like a problem with` E1`

Reply:
Has anyone noticed any problems with the `IkedaCarpenter` function that also uses the `exponentialIntegral` in `SpecialFunctionsSupport`?

Reply:
@RichardWaiteSTFC Thanks a lot for the comments. Here is what I have,

1. The first two commits were my fix for the E1 definition within `Bk2BkExpConvPV` but later on we decided to go with the one (`exponentialIntegral`) defined in `SpecialFunctionsSupport` and we ended up with 4 commits where the last two were for the second approach. So basically, if we can make `exponentialIntegral` in `SpecialFunctionsSupport`, we only need the last two commits here plus some other commits to fix the issue with `exponentialIntegral` in `SpecialFunctionsSupport`.

2. I will try to add more tests as you suggested.

3. Will make the release notes more specific.

4. Will definitely check through the two implementations and fix whatever issue with `exponentialIntegral` in `SpecialFunctionsSupport`.

5. I don't think people realized the issue with `exponentialIntegral` in `SpecialFunctionsSupport` before.

> You only see a problem on your branch when gamma > 0 - so definitely looking like a problem with E1

That is definitely true and that really makes the testing with non zero `gamma` necessary.

Reply:
@RichardWaiteSTFC Here is what I got,

1. There is no issue with `exponentialIntegral` in `SpecialFunctionsSupport` and therefore there is no issue with `IkedaCarpenterPV` function either. The problem is with my previous implementation of `exponentialIntegral` in `Bk2BkExpConvPV` - in the implementation for `exponentialIntegral` in `SpecialFunctionsSupport`, it is actually the following that is returned,

    ![image](https://user-images.githubusercontent.com/2872152/123334620-5b6d9280-d511-11eb-8483-e47805c6e39f.png)

    Therefore, the returned value is actually not the E1 function but with an extra multiplicative exponential term. In my previous implementation, I did not realize this and put another multiplicative exponential term in `Bk2BkExpConvPV` and therefore it causes the weird that you see in your example. Unfortunately, since originally we only had the test with gamma being 0 and therefore such a mistake was not caught at all.

2. More tests added.

    2.1. Compare with the output from `BackToBackExponential` when gamma is 0.

    2.2. Test for the situation where gamma is non-zero.

3. The new implementation is actually the same as my very original commit (where the separate version of E1 function defined in `Bk2BkExpConvPV` is used). I have double checked the new implementation by running exactly the same test as what I put in my original PR comments and the results are exactly the same.

4. Release docs changed to reflect a bit more details.